File: server/server/books/urls.py
"""Books urls"""
from django.urls import path

from server.books.views import CreateBook, DetailsBook, AllBooksByCategory, BookDeleteView, BookUpdateView, RateBook

urlpatterns = [
    path('create/', CreateBook.as_view(), name='create-book'),
    path('<str:book_status>/', AllBooksByCategory.as_view(), name='books-by-status'),
    path('details/<int:book_pk>/', DetailsBook.as_view(), name='details_book'),
    path('edit/<int:pk>/', BookUpdateView.as_view(), name='update_book'),
    path('delete/<int:id>/', BookDeleteView.as_view(), name='book-delete'),
    path('rate-book/<int:book_pk>/', RateBook.as_view(), name='book-rate'),
]

File: client/src/appPaths/urls.js
const baseUrl = 'http://127.0.0.1:8000/api/'

const urls = {
    Home: baseUrl,
    Login: `${baseUrl}login/`,
    Logout: `${baseUrl}logout/`,
    Register: `${baseUrl}register/`,
    MyBooks: `${baseUrl}my-books/`,
    BooksByStatus: (status) => `${baseUrl}books/${status}/`,
    BookDetail: (id) => `${baseUrl}books/details/${id}/`,
    BookEdit: (id) => `${baseUrl}books/edit/${id}/`,
    BookDelete: (id) => `${baseUrl}books/delete/${id}/`,
    RateBook: (id) => `${baseUrl}books/rate-book/${id}/`,
    AddBook: `${baseUrl}books/create/`,
    UserDetails: `${baseUrl}users/`,
    EditUser: `${baseUrl}users/update-profile/`,
    DeleteUser: `${baseUrl}users/delete/`,
  };
  
  export default urls;